spec/controllers/categories_controller_spec.rb

```diff
@@ -0,0 +1,203 @@
+      subject
+
+      expect(response).to have_http_status :ok
+      expect(json.count).to eq(1)
+      expect(json[0]['name']).to eq('Category')
+    end
+
+    include_examples 'Invalid JWT Token'
+  end
+
+  describe '#show' do
+    subject { get(category_path(slug), headers: { Authorization: jwt_token }) }
+
+    let(:slug) { current_category.slug }
+
+    context 'with valid params' do
+      it 'returns the given category' do
+        subject
+
+        expect(response).to have_http_status :ok
+        expect(json['name']).to eq(current_category.name)
+      end
+    end
+
+    context 'with invalid params' do
+      let(:slug) { 'invalid-slug' }
+
+      it 'returns errors' do
+        subject
+
+        expect(response).to have_http_status :not_found
+        expect(json['errors']).to include('Category not found')
+      end
+    end
+
+    context 'when category from another user' do
+      let(:slug) { another_category.slug }
+
+      it 'returns category not found' do
+        subject
+
+        expect(response).to have_http_status :not_found
+        expect(json['errors']).to include('Category not found')
+      end
+    end
+
+    include_examples 'Invalid JWT Token'
+  end
+
+  describe '#create' do
+    subject { post(categories_path, headers: { Authorization: jwt_token }, params: params) }
+
+    let(:name) { 'New category name' }
+    let(:params) do
+      {
+        name: name
+      }
+    end
+
+    it 'creates a new category' do
+      subject
+
+      expect(response).to have_http_status :created
+      expect(json['message']).to include('Category created with success')
+    end
+
+    context 'without name' do
+      let(:name) { nil }
+
+      it "returns name can't be blank error" do
+        subject
+
+        expect(json['errors']).to include("Name can't be blank")
+      end
+    end
+
+    describe 'validates name uniqueness' do
+      context 'when the user has category with same name' do
+        let!(:existing_category) { create(:category, user: current_user) }
+        let(:name) { existing_category.name }
+
+        it 'is not valid' do
+          subject
+
+          expect(json['errors']).to include('Name has already been taken')
+        end
+      end
+
+      context 'when another user has category with same name' do
+        let!(:another_user) { create(:user) }
+        let!(:existing_category) { create(:category, user: another_user) }
+        let(:name) { existing_category.name }
+
+        it 'is valid' do
+          subject
+
+          expect(json['message']).to include('Category created with success')
+        end
+      end
+    end
+
+    include_examples 'Invalid JWT Token'
+  end
+
+  describe '#update' do
+    subject { put(category_path(slug), headers: { Authorization: jwt_token }, params: params) }
+
+    let(:slug) { current_category.slug }
+    let(:name) { 'New category name' }
+    let(:params) do
+      {
+        name: name
+      }
+    end
```

Comment:
Se quiser deixar em 1 linha só:
```suggestion
    let(:params) { { name: name } }
```